### PR TITLE
Update prometheus

### PR DESCRIPTION
**Pull request by bboreham**
This is a re-run of #763, updated to address the broken assumption about which queriers are used where.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -19,6 +19,10 @@
   name = "github.com/prometheus/prometheus"
   branch = "master"
 
+[[constraint]]
```
Comment (juliusv) [inline Gopkg.toml]: Didn't we want to have comments for newly added constraints?

Comment (bboreham) [reply Gopkg.toml]: good spot

File: pkg/querier/querier.go [hunk 2]
```diff
@@ -2,40 +2,56 @@ package querier
...
 // ChunkStore is the interface we need to get chunks
 type ChunkStore interface {
 	Get(ctx context.Context, from, through model.Time, matchers ...*labels.Matcher) (model.Matrix, error)
 }
 
 // NewEngine creates a new promql.Engine for cortex.
-func NewEngine(distributor Querier, chunkStore ChunkStore) *promql.Engine {
-	queryable := NewQueryable(distributor, chunkStore, false)
-	return promql.NewEngine(queryable, nil)
+func NewEngine(distributor Querier, chunkStore ChunkStore, reg prometheus.Registerer, maxConcurrent int, timeout time.Duration) (*promql.Engine, storage.Queryable) {
```
Comment (juliusv) [inline pkg/querier/querier.go]: Having `NewEngine()` return both an engine and a queryable seems messy, or at least not appropriate naming anymore. Maybe just get rid of this tiny wrapper and have the caller call both `NewQueryable()` and `promql.NewEngine`?

Comment (bboreham) [reply pkg/querier/querier.go]: Agreed; I've done this.

**Comment by bboreham** [comment]
I've also re-applied the change described at https://github.com/weaveworks/cortex/pull/763#issuecomment-376247025 which I hadn't noticed before #763 was merged.

**Pull request merged**